src/SessionSetup/SessionSetupDialogTest.cpp

```diff
@@ -6,38 +6,81 @@
 
 #include <QApplication>
 #include <QPushButton>
+#include <QTest>
 #include <optional>
+#include <variant>
 
 #include "MetricsUploader/MetricsUploaderStub.h"
 #include "OrbitSsh/Context.h"
 #include "SessionSetup/Connections.h"
+#include "SessionSetup/DeploymentConfigurations.h"
 #include "SessionSetup/SessionSetupDialog.h"
+#include "SessionSetup/TargetConfiguration.h"
+#include "SessionSetup/TargetLabel.h"
 
 namespace orbit_session_setup {
 
-// TODO(b/189838600) Add tests that check correctness.
+namespace {
+orbit_metrics_uploader::MetricsUploaderStub uploader_stub{};
+}
 
-TEST(SessionSetupDialog, SmokeTest) {
+TEST(SessionSetupDialog, Exec) {
```

Comment:
I would prefer a more descriptive test name. Test names can be quite large, but ideally they should describe what gets tested and what is the expected output. 

In this case I would also split the test into two separate tests, CreateExecAndRejectEmptyDialogueReturnsNoConfiguration and CreateExecAndStartDialogueWithFileTargetReturnsValidConfiguration.
This goes a bit more into the direction of "behaviour tests".

Reply:
done :)

Reply:
I actually also added a test fixture to reuse code.. hence the fixup change has plenty changes

Reply:
Thanks! I had another look at the complete PR. LGTM.